File: SnappetChallenge/Models/Work.cs
using System;

namespace SnappetChallenge.Models
{
    public class Work
    {
        public Int64 SubmittedAnswerId { get; set; }
        public DateTime SubmitDateTime { get; set; }
        public int Correct { get; set; }
        public int Progress { get; set; }
        public Int64 UserId { get; set; }
        public Int64 ExerciseId { get; set; }
        public string Difficulty { get; set; }
        public string Subject { get; set; }
        public string Domain { get; set; }
        public string LearningObjective { get; set; }
    }
}

File: snappet-challenge/src/app/types/work.ts
export interface Work {
  SubmittedAnswerId: number;
  SubmitDateTime: string;
  Correct: number;
  Progress: number;
  UserId: number;
  ExerciseId: number;
  Difficulty: string;
  Subject: string;
  Domain: string;
  LearningObjective: string;
}

export interface LearningObjectivesDetails {
  [objectiveName: string]: {
    count: number;
  };
}
export interface DomainDetails {
  [domainName: string]: {
    count: number;
    objectives: LearningObjectivesDetails;
  };
}
export interface SubjectDetails {
  count: number;
  domains: DomainDetails;
}
export interface ProcessedWorkData {
  [subjectName: string]: SubjectDetails;
}
